state_management/javascript/sdk/order-processor/index.js

```diff
@@ -1,45 +1,58 @@
-import { DaprClient } from 'dapr-client'; 
+import { DaprClient } from '@dapr/dapr'
 
-const DAPR_HOST = process.env.DAPR_HOST || "http://localhost";
-const DAPR_HTTP_PORT = process.env.DAPR_HTTP_PORT || "3500";
+const DAPR_PROTOCOL = process.env.DAPR_PROTOCOL ?? "http"
+const DAPR_HOST = process.env.DAPR_HOST ?? "localhost"
+
+let PORT
+switch (DAPR_PROTOCOL) {
+    case "http": {
+        PORT = process.env.DAPR_HTTP_PORT
+        break
+    }
+    case "grpc": {
+        PORT = process.env.DAPR_GRPC_PORT
+        break
+    }
+    default: {
+        PORT = 3500
+    }
+}
+
+const DAPR_STATE_STORE_NAME = "statestore"
 
 async function main() {
-    for(var i = 1; i <= 10; i++) {
-        const orderId = i;
-        const order = {orderId: orderId};
-        const client = new DaprClient(DAPR_HOST, DAPR_HTTP_PORT);
-        const STATE_STORE_NAME = "statestore";
-
-        // Save state into the state store
-        client.state.save(STATE_STORE_NAME, [
+    const host = `${DAPR_PROTOCOL}://${DAPR_HOST}`
```

Comment:
You don't need to use the host here as we will automatically do this for you (https://github.com/dapr/js-sdk/blob/master/src/implementation/Client/HTTPClient/HTTPClient.ts#L47) so you can drop the url creation here and just pass `localhost`.


As for which protocol to use, maybe you can just use: 

```javascript
import { DaprClient, CommunicationProtocol } from "@dapr/dapr";

const protocol = (DAPR_PROTOCOL === "http") ? CommunicationProtocol.HTTP : CommunicationProtocol.GRPC;
const client = new DaprClient(host, PORT, protocol); 
```

Reply:
Ok, I updated to use something similar to your suggestion

- Use `CommunicationProtocolEnum` instead of 'http' string
- Remove protocol from url construction and instead pass protocol to DaprClient constructor
- use lowercase version of variables (port, host, etc) since they were not constants
- Remove `orderId` to avoid confusion about why it wasn't used

Given DAPR_PROTOCOL is non standard and is NOT set automatically by the Dapr CLI like the other env vars we have to assume it will not be set and in that case default both to "http"

### Ideas for Dapr / CLI improvement

Given this determining of which port / env var to use based on protocol is going to be common across all Dapr apps. I think there should be first-class support for this in the Dapr SDK and the Dapr CLI should make DAPR_PROTOCOL standard and set it just like DAPR_GRPC_PORT and DAPR_HTTP_PORT

option | var
--- | ---
--app-protocol http | ? (why is nothing set for this?)
--dapr-http-port 3500 | DAPR_HTTP_PORT
--dapr-grpc-port 6001 | DAPR_GRPC_PORT